### PR TITLE
Feat/105/secondary indexes inputs outputs

### LINKED ISSUE #105: Consider using secondary indexes in some queries
Consider using secondary indexes for
- [x] `get_spent`
- [x] `get_owned_ids`
- [x] `get_transaction`

### LINKED ISSUE #787: Performance of get_owned_ids
Extract from Gitter:

> **Roderik van der Veer @roderik Nov 02 20:29**:
> Is anyone experiencing very slow “get_owned_ids”? I've got 5 addresses with combined 80 assets in them, and when i do get_owned_ids on each of them (concurrently) i still end up with about 4.5 seconds to get all the results back. This would totally suck if i put in 100k assets 
I’m still on 0.6 btw

> **Rodolphe Marques @r-marques 09:22**:
> @roderik can you open a github issue. We should consider creating a secondary index for this query

### PR DESCRIPTION
**Pull request by r-marques**
resolves #105 
partially resolves #787

This pr adds secondary indexes to the `public_keys` of `outputs` used by `get_owned_ids` and `fulfills` in the `inputs` used by `get_spent`.

### CODE REVIEW

**Comment by libscott** [comment]
Looks nice

Would be cool to have some tests that verify which indexes are being used for a query at some point.

**Comment by r-marques** [comment]
Create a separate issue to test that secondary indexes are being used #1126

File: bigchaindb/backend/rethinkdb/schema.py [hunk 1]
```diff
@@ -66,6 +66,31 @@ def create_bigchain_secondary_index(connection, dbname):
...
+    # `public_keys: [pk1, pk2, pk3]`
+    connection.run(
+        r.db(dbname)
+         .table('bigchain')
+         .index_create('outputs',
+                       r.row['block']['transactions']
+                        .concat_map(lambda tx: tx['outputs']['public_keys'])
+                        .reduce(lambda l, r: l + r), multi=True))
+
+    # secondary index on inputs/transaction links (txid, cid)
```
Comment (sohkai) [inline bigchaindb/backend/rethinkdb/schema.py]: Maybe use `output id` instead of `cid`.

**Comment by libscott** [comment]
Yea. Actually the rethinkdb one is powerful.

**Pull request merged**